### PR TITLE
Implement basic build process

### PR DESCRIPTION
**Pull request by davidmohar**
Adding GitHub Actions build for some basic sanity before deploying & merging

### CODE REVIEW

**Comment by davidmohar** [comment]
@1234filip can you check build errors and fix them in scope of this PR?

File: world-server/src/network/Server.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import * as http from 'http';
 import * as websocket from 'websocket';
-import { Logger, Messages, Router } from 'world-core';
+import { Logger, Messages, Router } from '../../../world-core';
```
Comment (davidmohar) [inline world-server/src/network/Server.ts]: This is not OK, you should always import from `world-core` and expose the types there. Did you forget to `yarn core build`?

File: world-server/src/network/Server.ts [hunk 2]
```diff
@@ -12,6 +12,10 @@ export class Server implements Messages.MsgHandler {
     constructor() {
         this.connCounter = 0;
         this.sockets = [];
+
+        // Only to bypass the build errors
+        this.httpServer = http.createServer();
+        this.wsServer = new websocket.server();
```
Comment (davidmohar) [inline world-server/src/network/Server.ts]: There is probably a way better solution for this then instantiating dummy servers

File: world-server/src/network/Server.ts [hunk 3]
```diff
@@ -50,10 +54,13 @@ export class Server implements Messages.MsgHandler {
         var connection = request.accept('', request.origin);
         Logger.info(`Accepted socket connection from ${request.remoteAddress}`)
 
-        connection.internalId = this.connCounter++;
+        // Commented out for now to avoid build errors
+        // the library should do this automatically if debug is enabled
+        // https://github.com/theturtle32/WebSocket-Node/blob/1f7ffba2f7a6f9473bcb39228264380ce2772ba7/lib/WebSocketConnection.js#L42
+        // connection.internalId = this.connCounter++;
```
Comment (davidmohar) [inline world-server/src/network/Server.ts]: We will need to track our internal ID on the connection itself so we must find a solution for this problem. I don't want to wrap this into our own class. Can we do something like
```typescript
connection['internalId'] = this.connCounter++;
```

Comment (filiptrplan) [reply world-server/src/network/Server.ts]: I looked at a couple of things but TypeScript disallows this kind of dynamic property adding. The only solution I see, if we really want to use the `WebSocketConnection` class, is that we fork the repo and add our own `internalId` to the class.

File: world-server/src/network/Server.ts (hunk 1 above)
Comment (filiptrplan) [reply world-server/src/network/Server.ts]: Even after `yarn core build` the error persists. Any suggestions?
UPDATE: had to re-run `yarn install`

File: world-server/src/network/Server.ts (hunk 2 above)
Comment (filiptrplan) [reply world-server/src/network/Server.ts]: The alternative is that we call `init()` in the constructor, which doesn't seem like a bad idea for our use case. I'll implement it and  let me know if this shouldn't be done.

Comment (filiptrplan) [reply world-server/src/network/Server.ts]: UPDATE: This solution doesn't work, reverting to the dummy server solution for now.

**Comment by 1234filip** [comment]
I think the build error is due to not running `yarn install` on the CI check. This solved this exact build error for me. Could you check it out @davidmohar

**Review by filiptrplan — DISMISSED** [review]
Looks good :)

**Review by aJuvan — DISMISSED** [review]

**Review by jakakonda — APPROVED** [review]

**Pull request merged**